text/0000-integer_atomics.md

```diff
@@ -0,0 +1,105 @@
+- Feature Name: integer_atomics
+- Start Date: 2016-03-14
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC basically changes `core::sync::atomic` to look like this:
+
+```rust
+#[cfg(target_has_atomic = "8")]
+struct AtomicBool {}
+#[cfg(target_has_atomic = "8")]
+struct AtomicI8 {}
+#[cfg(target_has_atomic = "8")]
+struct AtomicU8 {}
+#[cfg(target_has_atomic = "16")]
+struct AtomicI16 {}
+#[cfg(target_has_atomic = "16")]
+struct AtomicU16 {}
+#[cfg(target_has_atomic = "32")]
+struct AtomicI32 {}
+#[cfg(target_has_atomic = "32")]
+struct AtomicU32 {}
+#[cfg(target_has_atomic = "64")]
+struct AtomicI64 {}
+#[cfg(target_has_atomic = "64")]
+struct AtomicU64 {}
+#[cfg(target_has_atomic = "128")]
+struct AtomicI128 {}
+#[cfg(target_has_atomic = "128")]
+struct AtomicU128 {}
+#[cfg(target_has_atomic = "ptr")]
+struct AtomicIsize {}
+#[cfg(target_has_atomic = "ptr")]
+struct AtomicUsize {}
+#[cfg(target_has_atomic = "ptr")]
+struct AtomicPtr<T> {}
```

Comment:
@brson are you stating that because we've previously promised support for atomic ptr-size values, we're stuck supporting them for all platforms rust targets for the foreseeable future? In effect: if we ever want to support a platform without pointer-size atomics, it will require compiler-rt provide some AtomicPtr work-alike?

As the RFC notes later, all architectures we currently support would be unaffected. Only new, not yet existent archs would be broken. I'm presuming you don't think this is sufficient?

Is there some other process you're implicitly proposing here that I'm missing?